logs/handler.go

```diff
@@ -0,0 +1,431 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/openfaas/faas-provider/httputils"
+)
+
+var upgrader = websocket.Upgrader{} // use default options
+
+// Requestor submits queries the logging system.
+// This will be passed to the log handler constructor.
+type Requestor interface {
+	// Query submits a log request to the actual logging system.
+	Query(context.Context, Request) (<-chan Message, error)
+}
+
+// NewSimpleLogHandlerFunc creates and http HandlerFunc from the supplied log Requestor.
+func NewSimpleLogHandlerFunc(requestor Requestor) http.HandlerFunc {
```

Comment:
@stefanprodan This is what code looks like when WriteTimeout is *disabled* on the http.Server.